base/broadcast.jl

```diff
@@ -110,6 +139,9 @@ const bitcache_size = 64 * bitcache_chunks # do not change this
 dumpbitcache(Bc::Vector{UInt64}, bind::Int, C::Vector{Bool}) =
     Base.copy_to_bitarray_chunks!(Bc, ((bind - 1) << 6) + 1, C, 1, min(bitcache_size, (length(Bc)-bind+1) << 6))
 
+@inline _broadcast_getvals(A, I) = A
+@inline _broadcast_getvals(A::Union{AbstractArray,Tuple}, I) = A[I]
```

Comment:
I thought of that, but there is already `broadcast_getindex`, although it doesn't seem to be used anywhere in Base. I'm not sure of its uses outside Base neither.